### PR TITLE
[py-smart-open] fixing dependencies

### PR DESCRIPTION
**Pull request by qwertos**

### CODE REVIEW

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Please merge develop and re-push (so `py-boto` picked up from previous PR).  This is needed for the tests to pass, specifically for `test_all_dependencies_exist`'s catching the missing `py-boto` dependency.

File: var/spack/repos/builtin/packages/py-smart-open/package.py [hunk 1]
```diff
@@ -21,5 +21,6 @@ class PySmartOpen(PythonPackage):
     depends_on('py-setuptools', type='build')
     depends_on('py-requests', type=('build', 'run'))
     depends_on('py-boto3', type=('build', 'run'))
-    depends_on('py-google-cloud-storage', type=('build', 'run'))
+    depends_on('py-boto', when='@1.8.4', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-smart-open/package.py]:
```suggestion
    depends_on('py-boto@2.32:', when='@1.8.4', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
LGTM.  I am able to successfully build the package.

**Comment by tldahlgren** [comment]
Thank you for your contribution.

**Pull request merged**